File: api/src/main/java/com/ruby/rubia_server/core/entity/MessageTemplateRevision.java
package com.ruby.rubia_server.core.entity;

import com.ruby.rubia_server.core.base.BaseEntity;
import com.ruby.rubia_server.core.enums.RevisionType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "message_template_revisions")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MessageTemplateRevision implements BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company; // Para multi-tenancy

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "template_id", nullable = false)
    private MessageTemplate template; // Template ao qual esta revisão pertence

    @Column(nullable = false)
    private Integer revisionNumber; // Número da revisão (1 para a original, 2 para a primeira edição, etc.)

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content; // Conteúdo desta revisão específica do template

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "edited_by_user_id")
    private User editedBy; // Quem criou/editou esta revisão

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "revision_type", nullable = false)
    @Builder.Default
    private RevisionType revisionType = RevisionType.EDIT; // Tipo da revisão (CREATE, EDIT, DELETE, RESTORE)

    // Novos campos para metadados de IA
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ai_agent_id")
    private AIAgent aiAgent; // Agente de IA usado para esta melhoria (se aplicável)

    @Column(name = "ai_enhancement_type")
    private String aiEnhancementType; // Tipo de melhoria aplicada pela IA (friendly, professional, etc.)

    @Column(name = "ai_tokens_used")
    private Integer aiTokensUsed; // Tokens consumidos pela IA

    @Column(name = "ai_credits_consumed")
    private Integer aiCreditsConsumed; // Créditos consumidos pela IA

    @Column(name = "ai_model_used")
    private String aiModelUsed; // Nome do modelo de IA usado

    @Column(columnDefinition = "TEXT", name = "ai_explanation")
    private String aiExplanation; // Explicação das melhorias aplicadas pela IA

    @CreationTimestamp
    @Column(name = "revision_timestamp", updatable = false)
    private LocalDateTime revisionTimestamp; // Quando esta revisão foi criada

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/Entities/MessageTemplateRevision.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("message_template_revisions")]
public class MessageTemplateRevision : BaseEntity
{
    [Required]
    [Column("revision_number")]
    public int RevisionNumber { get; set; } // Número da revisão (1 para a original, 2 para a primeira edição, etc.)

    [Required]
    [Column("content", TypeName = "TEXT")]
    public string Content { get; set; } = string.Empty; // Conteúdo desta revisão específica do template

    [Column("revision_type")]
    [Required]
    public RevisionType RevisionType { get; set; } = RevisionType.Edit; // Tipo da revisão

    // Novos campos para metadados de IA
    [Column("ai_enhancement_type")]
    public string? AiEnhancementType { get; set; } // Tipo de melhoria aplicada pela IA

    [Column("ai_tokens_used")]
    public int? AiTokensUsed { get; set; } // Tokens consumidos pela IA

    [Column("ai_credits_consumed")]
    public int? AiCreditsConsumed { get; set; } // Créditos consumidos pela IA

    [Column("ai_model_used")]
    public string? AiModelUsed { get; set; } // Nome do modelo de IA usado

    [Column("ai_explanation", TypeName = "TEXT")]
    public string? AiExplanation { get; set; } // Explicação das melhorias aplicadas pela IA

    [Column("revision_timestamp")]
    [Required]
    public DateTime RevisionTimestamp { get; set; } = DateTime.UtcNow; // Quando esta revisão foi criada

    // Navigation properties
    [Column("company_id")]
    [Required]
    public Guid CompanyId { get; set; }
    
    [ForeignKey("CompanyId")]
    public virtual Company Company { get; set; } = null!;

    [Column("template_id")]
    [Required]
    public Guid TemplateId { get; set; }
    
    [ForeignKey("TemplateId")]
    public virtual MessageTemplate Template { get; set; } = null!; // Template ao qual esta revisão pertence

    [Column("edited_by_user_id")]
    public Guid? EditedByUserId { get; set; }
    
    [ForeignKey("EditedByUserId")]
    public virtual User? EditedBy { get; set; } // Quem criou/editou esta revisão

    [Column("ai_agent_id")]
    public Guid? AiAgentId { get; set; }
    
    [ForeignKey("AiAgentId")]
    public virtual AIAgent? AiAgent { get; set; } // Agente de IA usado para esta melhoria (se aplicável)
}